terraform/groups/provisioning-role/module-iam-policies/terraform-apply.tf

```diff
@@ -0,0 +1,34 @@
+data "aws_iam_policy_document" "apply" {
+  statement {
+    sid       = "ApplyPolicyAllResources"
+    effect    = "Allow"
+    resources = ["*"]
+    actions   = [
+      "ec2:CreateSecurityGroup",
+      "ec2:CreateTags",
+      "lambda:CreateFunction",
+    ]
+  }
+
+  statement {
+    sid       = "ApplyPolicyListedResources"
+    effect    = "Allow"
+    resources = [
+      "arn:aws:iam::${var.aws_account_id}:role/payment-reconciler-execution-*",
+      "arn:aws:events:eu-west-2:${var.aws_account_id}:rule/payment-reconciler-*",
+      "arn:aws:ec2:eu-west-2:${var.aws_account_id}:security-group/*",
```

Comment:
Would this not allow the action to be done on ALL security groups in the account, not just payment-reconciler ones

Reply:
Yes, this is the minefield I was chatting to Jimmy about. The problem I found with the action `ec2:CreateSecurityGroup` is that it doesn't allow any restrictions on resources.
I was able to add a condition for `ec2:DeleteSecurityGroup` (in the destroy policy) to restrict on the name, but the Create equivalent has no `ec2:ResourceTag` condition key.
Let me know if you can think of some other ways around this.